Sources/VitalHealthKit/HealthKit/VitalHealthKitClient.swift

```diff
@@ -224,6 +224,25 @@ extension VitalHealthKitClient {
 
     self.backgroundDeliveryTask = Task(priority: .high) {
       for await payload in stream {
+        // If the task is cancelled, we would break the endless iteration and end the task.
+        // Any buffered payload would not be processed, and is expected to be redelivered by
+        // HealthKit.
+        //
+        // https://developer.apple.com/documentation/healthkit/hkhealthstore/1614175-enablebackgrounddelivery
+        try Task.checkCancellation()
+
+        // Task is not cancelled — we must call the HealthKit completion handler irrespective of
+        // the sync process outcome. This is to avoid triggering the "strike on 3rd missed delivery"
+        // rule of HealthKit background delivery.
+        //
+        // Since we have fairly frequent delivery anyway, each of which will implicit retry from
+        // where the last sync has left off, this unconfigurable exponential backoff retry
+        // behaviour adds little to no value in maintaining data freshness.
+        //
+        // (except for the task cancellation redelivery expectation stated above).
+        defer { payload.completion() }
```

Comment:
For some reason it feels wrong to have this defer here. Shouldn't this be called at the end of the loop?

Reply:
We have 3 branches below and all of them should call completion. `defer` is the most suitable language construct besides manually inserting the call 3 times.

Reply:
whichever point that ends the iteration

Reply:
Oh I see.